samplePDF/samplePDFFDBase.cpp

```diff
@@ -340,328 +400,289 @@ void samplePDFFDBase::fillArray() {
       //The alternative would be to have inheritance based on BinningOpt
       double XVar = *(MCSamples[iSample].x_var[iEvent]);
-
+      
       //DB Find the relevant bin in the PDF for each event
       int XBinToFill = -1;
       int YBinToFill = MCSamples[iSample].NomYBin[iEvent];
-
+      
       //DB - First, check to see if the event is still in the nominal bin
       if (XVar < MCSamples[iSample].rw_upper_xbinedge[iEvent] && XVar >= MCSamples[iSample].rw_lower_xbinedge[iEvent]) {
-		XBinToFill = MCSamples[iSample].NomXBin[iEvent];
+	XBinToFill = MCSamples[iSample].NomXBin[iEvent];
       }
       //DB - Second, check to see if the event is outside of the binning range and skip event if it is
-	  //ETA- note that nXBins is XBinEdges.size() - 1
+      //ETA- note that nXBins is XBinEdges.size() - 1
       else if (XVar < XBinEdges[0] || XVar >= XBinEdges[nXBins]) {
-		continue;
+	continue;
       }
       //DB - Thirdly, check the adjacent bins first as Eb+CC+EScale shifts aren't likely to move an Erec more than 1bin width
       //Shifted down one bin from the event bin at nominal
       else if (XVar < MCSamples[iSample].rw_lower_xbinedge[iEvent] && XVar >= MCSamples[iSample].rw_lower_lower_xbinedge[iEvent]) {
-		XBinToFill = MCSamples[iSample].NomXBin[iEvent]-1;
+	XBinToFill = MCSamples[iSample].NomXBin[iEvent]-1;
       }
       //Shifted up one bin from the event bin at nominal
       else if (XVar < MCSamples[iSample].rw_upper_upper_xbinedge[iEvent] && XVar >= MCSamples[iSample].rw_upper_xbinedge[iEvent]) {
-		XBinToFill = MCSamples[iSample].NomXBin[iEvent]+1;
+	XBinToFill = MCSamples[iSample].NomXBin[iEvent]+1;
       }
       //DB - If we end up in this loop, the event has been shifted outside of its nominal bin, but is still within the allowed binning range
-	  else {
-		for (unsigned int iBin=0;iBin<(XBinEdges.size()-1);iBin++) {
-		  if (XVar >= XBinEdges[iBin] && XVar < XBinEdges[iBin+1]) {
-			XBinToFill = iBin;
-		  }
-		}
+      else {
+	for (unsigned int iBin=0;iBin<(XBinEdges.size()-1);iBin++) {
+	  if (XVar >= XBinEdges[iBin] && XVar < XBinEdges[iBin+1]) {
+	    XBinToFill = iBin;
 	  }
-
+	}
+      }
+      
       //DB Fill relevant part of thread array
       if (XBinToFill != -1 && YBinToFill != -1) {
-		//std::cout << "Filling samplePDFFD_array at YBin: " << YBinToFill << " and XBin: " << XBinToFill << std::endl;
-		samplePDFFD_array[YBinToFill][XBinToFill] += totalweight;
-		samplePDFFD_array_w2[YBinToFill][XBinToFill] += totalweight*totalweight;
+	samplePDFFD_array[YBinToFill][XBinToFill] += totalweight;
+	samplePDFFD_array_w2[YBinToFill][XBinToFill] += totalweight*totalweight;
       }
     }
   }
-
+  
 #endif // end the else in openMP
   return;
 }
 
 #ifdef MULTITHREAD
+// ************************************************ 
+/// Multithreaded version of fillArray @see fillArray()
+// ************************************************ 
 void samplePDFFDBase::fillArray_MP() 
```

Comment:
this is so similar to the 'single threaded' version, I think it would be worth trying to reduce code duplication